addons/website/static/src/components/navbar/navbar.js

```diff
@@ -9,70 +10,112 @@ import { OptimizeSEODialog } from '@website/components/dialog/seo';
+                isDisplayed: () => canShowPageProperties(),
                 getProps: () => ({
                     onRecordSaved: (record) => {
-                        return this.orm.read('website.page', [record.resId], ['url']).then(res => {
-                            this.websiteService.goToWebsite({websiteId: record.data.website_id[0], path: res[0]['url']});
+                        return orm.read('website.page', [record.resId], ['url']).then(res => {
+                            website.goToWebsite({websiteId: record.data.website_id[0], path: res[0]['url']});
                         });
                     },
                 })
             },
         };
-    },
 
-    filterWebsiteMenus(sections) {
-        const filteredSections = [];
-        for (const section of sections) {
-            const isWebsiteCustomMenu = this.websiteEditingMenus[section.xmlid];
-            const displayWebsiteCustomMenu = isWebsiteCustomMenu && this.websiteService.isRestrictedEditor && this.websiteEditingMenus[section.xmlid].isDisplayed();
-            if (!isWebsiteCustomMenu || displayWebsiteCustomMenu) {
-                let subSections = [];
-                if (section.childrenTree.length) {
-                    subSections = this.filterWebsiteMenus(section.childrenTree);
+        return {
+            get menus() {
+                return menus;
+            },
+            get(xmlId) {
+                return this.menus[xmlId];
+            },
+            open(xmlId) {
+                const menu = this.get(xmlId);
+                if (menu.openWidget) {
+                    return menu.openWidget();
                 }
-                filteredSections.push(Object.assign({}, section, {childrenTree: subSections}));
-            }
+                return dialog.add(
+                    menu.Component,
+                    menu.getProps && menu.getProps(),
+                );
+            },
+            addCustomMenus(sections) {
+                const filteredSections = [];
+                for (const section of sections) {
+                    const isWebsiteCustomMenu = !!this.get(section.xmlid);
+                    const displayWebsiteCustomMenu = isWebsiteCustomMenu && website.isRestrictedEditor && this.get(section.xmlid).isDisplayed();
+                    if (!isWebsiteCustomMenu || displayWebsiteCustomMenu) {
+                        let subSections = [];
+                        if (section.childrenTree.length) {
+                            subSections = this.addCustomMenus(section.childrenTree);
+                        }
+                        filteredSections.push(Object.assign({}, section, {childrenTree: subSections}));
+                    }
+                }
+                return filteredSections;
+            },
+        };
+    }
+};
+
+registry.category('services').add('website_custom_menus', websiteCustomMenus);
+
+patch(NavBar.prototype, 'website_navbar', {
```

Comment:
I guess the patch is something we want to get rid of in the long term, right? Maybe add another commit in your PR renaming/moving your file as a service as it is the main purpose of this file? I think I would rather split it in two actually.

Try to do this:
- Rename the file as a service
- Make your current diff
- Move the patches in another file (losing their history, but thus preserving the service part's history).